### ISSUE TITLE
Implementing "appman update" for non-updatable AppImages

### ISSUE THREAD

**Issue by ivan-hc** [ivan-hc/AppMan#2]
Hi folks,
I'm planning a solution for the bigger issue of AppMan: the updates.

We have a lot of AppImages without inbuilt instructions about furter updates, and for now I've tried to solve this situation with the reinstallation of the program with `appman -i <program>`, but it is not sufficient enough to made the final user aware on the availability of new versions of the software.

However, by typing `<program> --help` on some AppImages, it is possible to see the actual version installed.

My idea is this, I want to implement a script in the app's directory to call with the `appman -u` command only if <program> is in /opt/bin, after checked the version's number for of that program, and here are two cases:

1. if the output of `<program> --help` has the same value of the program available in its source, no action needed;
2. if the output is lower than a new version of the software in the source, do `appman -i <program>`.

Obviously the script must contain the link to the source of the package to do so.

There is a lot of work to do, if I'll have time I'll try to do some experiments. Any idea is welcome.

PS: I can already implement a solution for those AppImages with an external .zsync file (for example Libreoffice), but for now the main goal is to write a script for this issue first, then downloading .zsync files will be easier.

**Comment by ivan-hc** [comment]
TEST 1 on "about-$APP"

```
URL2="https://raw.githubusercontent.com/ivan-hc/AppMan/main/applications"
cd /opt/bin && for f in *; do wget -q $URL2/"$f"/about-"$f" && cat ./about-"$f"; echo ""; rm ./about-"$f"; done
```
SUCCESSFUL!

Now I have to implement scripts for each application.

**Comment by ivan-hc** [comment]
Today I've added the -U (full-upgrade) option ([commit](https://github.com/ivan-hc/AppMan/commit/7273a784a2b7bb885bbee263da3eef74fb525d04)), this will re-install all the existing programs on your pc. It is not much, but with the re-installation of all programs (maybe one day per month) you can be sure that all programs are updated.
This workaround is different from what I wanted to do, but for now it's more than enough to fix the issue of updates for non-updateable apps (which are largely official AppImage, sad but true). At least in this way I can continue to add applications and doing researches.

**Comment by ivan-hc** [comment]
Problem solved with the [AppMan 1.5](https://github.com/ivan-hc/AppMan/releases/tag/1.5) release.
